AssertionError when using models.Manager

# Bug report

<!--
Hi, thanks for submitting a bug. We appreciate that.

But, we will need some information about what's wrong to help you.
-->

## What's wrong
I can't seem to create a minimal example demonstrating this, but I'm getting this on one of my projects.

If I change a custom queryset to become a custom manager by changing what it inherits from and changing `objects = CustomQuerySet.as_manager()` to `objects = CustomManager()` I get the following.

```
Traceback (most recent call last):
  File "c:\users\dusti\.pyenv\pyenv-win\versions\3.8.0-amd64\lib\runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
    exec(code, run_globals)
  File "mypy\semanal.py", line 4679, in accept
  File "mypy\nodes.py", line 939, in accept
  File "mypy\semanal.py", line 1001, in visit_class_def
  File "mypy\semanal.py", line 1078, in analyze_class
  File "mypy\semanal.py", line 1087, in analyze_class_body_common
  File "mypy\semanal.py", line 1147, in apply_class_plugin_hooks
  File "c:\users\dusti\appdata\local\pypoetry\cache\virtualenvs\vehicles-caet0sxz-py3.8\lib\site-packages\mypy_django_plugin\main.py", line 37, in transform_model_class
    process_model_class(ctx, django_context)
  File "c:\users\dusti\appdata\local\pypoetry\cache\virtualenvs\vehicles-caet0sxz-py3.8\lib\site-packages\mypy_django_plugin\transformers\models.py", line 357, in process_model_class
    initializer_cls(ctx, django_context).run()
  File "c:\users\dusti\appdata\local\pypoetry\cache\virtualenvs\vehicles-caet0sxz-py3.8\lib\site-packages\mypy_django_plugin\transformers\models.py", line 71, in run
    self.run_with_model_cls(model_cls)
  File "c:\users\dusti\appdata\local\pypoetry\cache\virtualenvs\vehicles-caet0sxz-py3.8\lib\site-packages\mypy_django_plugin\transformers\models.py", line 230, in run_with_model_cls
    custom_manager_type = self.create_new_model_parametrized_manager(custom_model_manager_name,
  File "c:\users\dusti\appdata\local\pypoetry\cache\virtualenvs\vehicles-caet0sxz-py3.8\lib\site-packages\mypy_django_plugin\transformers\models.py", line 183, in create_new_model_parametrized_manager
    helpers.copy_method_to_another_class(new_cls_def_context,
  File "c:\users\dusti\appdata\local\pypoetry\cache\virtualenvs\vehicles-caet0sxz-py3.8\lib\site-packages\mypy_django_plugin\lib\helpers.py", line 346, in copy_method_to_another_class
    assert bound_return_type is not None
AssertionError:
```

- OS: Win10
- `python` version: 3.8
- `django` version: 3.0.6
- `mypy` version: .770
- `django-stubs` version: 1.5.0

I've experienced a similar traceback. I've created a reproduction repo: https://github.com/danifus/djstubs-reproduce

Steps:
- `cd` to the directory containing `mypy.ini`
- run `mypy . --tb`

Gives me:
```
./test_app/models.py:10: error: INTERNAL ERROR -- Please try using mypy master on Github:
https://mypy.rtfd.io/en/latest/common_issues.html#using-a-development-mypy-build
Please report a bug at https://github.com/python/mypy/issues
version: 0.770
Traceback (most recent call last):
  File "mypy/semanal.py", line 4679, in accept
  File "mypy/nodes.py", line 939, in accept
  File "mypy/semanal.py", line 1001, in visit_class_def
  File "mypy/semanal.py", line 1078, in analyze_class
  File "mypy/semanal.py", line 1087, in analyze_class_body_common
  File "mypy/semanal.py", line 1147, in apply_class_plugin_hooks
  File "<venv_dir>/mypytest/lib/python3.7/site-packages/mypy_django_plugin/main.py", line 37, in transform_model_class
    process_model_class(ctx, django_context)
  File "<venv_dir>/mypytest/lib/python3.7/site-packages/mypy_django_plugin/transformers/models.py", line 357, in process_model_class
    initializer_cls(ctx, django_context).run()
  File "<venv_dir>/mypytest/lib/python3.7/site-packages/mypy_django_plugin/transformers/models.py", line 71, in run
    self.run_with_model_cls(model_cls)
  File "<venv_dir>/mypytest/lib/python3.7/site-packages/mypy_django_plugin/transformers/models.py", line 231, in run_with_model_cls
    base_manager_info=manager_info)
  File "<venv_dir>mypytest/lib/python3.7/site-packages/mypy_django_plugin/transformers/models.py", line 186, in create_new_model_parametrized_manager
    method_node=sym.node)
  File "<venv_dir>/mypytest/lib/python3.7/site-packages/mypy_django_plugin/lib/helpers.py", line 355, in copy_method_to_another_class
    assert bound_arg_type is not None
AssertionError:
./test_app/models.py:10: : note: use --pdb to drop into pdb
```

Let me know if you need anything else to help this along.

--- a/mypy_django_plugin/lib/helpers.py
+++ b/mypy_django_plugin/lib/helpers.py
@@ -343,6 +343,7 @@ def copy_method_to_another_class(ctx: ClassDefContext, self_type: Instance,
     arguments = []
     bound_return_type = semanal_api.anal_type(method_type.ret_type,
                                               allow_placeholder=True)
+
     assert bound_return_type is not None
 
     if isinstance(bound_return_type, PlaceholderNode):
@@ -352,8 +353,13 @@ def copy_method_to_another_class(ctx: ClassDefContext, self_type: Instance,
                                                      method_type.arg_types[1:],
                                                      method_node.arguments[1:]):
         bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True)
-        assert bound_arg_type is not None
+       
+        if bound_arg_type is None and not semanal_api.final_iteration:
+            semanal_api.defer()
+            return
 
+        assert bound_arg_type is not None
+        
         if isinstance(bound_arg_type, PlaceholderNode):
             return
 

--- a/mypy_django_plugin/lib/helpers.py
+++ b/mypy_django_plugin/lib/helpers.py
@@ -352,7 +352,7 @@ def copy_method_to_another_class(ctx: ClassDefContext, self_type: Instance,
     for arg_name, arg_type, original_argument in zip(method_type.arg_names[1:],
                                                      method_type.arg_types[1:],
                                                      method_node.arguments[1:]):
-        bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True)       
+        bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True) 
         if bound_arg_type is None and not semanal_api.final_iteration:
             semanal_api.defer()
             return

--- a/mypy_django_plugin/lib/helpers.py
+++ b/mypy_django_plugin/lib/helpers.py
@@ -352,7 +352,7 @@ def copy_method_to_another_class(ctx: ClassDefContext, self_type: Instance,
     for arg_name, arg_type, original_argument in zip(method_type.arg_names[1:],
                                                      method_type.arg_types[1:],
                                                      method_node.arguments[1:]):
-        bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True) 
+        bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True)
         if bound_arg_type is None and not semanal_api.final_iteration:
             semanal_api.defer()
             return

--- a/mypy_django_plugin/lib/helpers.py
+++ b/mypy_django_plugin/lib/helpers.py
@@ -352,14 +352,13 @@ def copy_method_to_another_class(ctx: ClassDefContext, self_type: Instance,
     for arg_name, arg_type, original_argument in zip(method_type.arg_names[1:],
                                                      method_type.arg_types[1:],
                                                      method_node.arguments[1:]):
-        bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True)
-       
+        bound_arg_type = semanal_api.anal_type(arg_type, allow_placeholder=True)       
         if bound_arg_type is None and not semanal_api.final_iteration:
             semanal_api.defer()
             return
 
         assert bound_arg_type is not None
-        
+
         if isinstance(bound_arg_type, PlaceholderNode):
             return
 

--- a/test-data/typecheck/managers/test_managers.yml
+++ b/test-data/typecheck/managers/test_managers.yml
@@ -335,3 +335,25 @@
                     objects = MyManager()
                 class ChildUser(models.Model):
                     objects = MyManager()
+
+-   case: custom_manager_annotate_method_before_type_declaration 
+    main: |
+        from myapp.models import ModelA, ModelB, ManagerA
+        reveal_type(ModelA.objects)  # N: Revealed type is 'myapp.models.ModelA_ManagerA1[myapp.models.ModelA]'
+        reveal_type(ModelA.objects.do_something)  # N: Revealed type is 'def (other_obj: myapp.models.ModelB) -> builtins.str'
+    installed_apps:
+        - myapp
+    files:
+        -   path: myapp/__init__.py
+        -   path: myapp/models.py
+            content: |
+                from django.db import models
+                class ManagerA(models.Manager):
+                    def do_something(self, other_obj: "ModelB") -> str:
+                        return 'test'
+                class ModelA(models.Model):
+                    title = models.TextField()
+                    objects = ManagerA()
+                class ModelB(models.Model):
+                    movie = models.TextField()
+

